test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1

```diff
@@ -0,0 +1,129 @@
+﻿function Write-LogPassThru
+{
+    Param(
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true, Position = 0)]
+        [string] $Message,
+        $Path = "$env:Temp\CodeCoverageRunLogs.txt"
+    )
+
+    $message = "$([Datetime]::Now.ToShortDateString()) - $([datetime]::Now.ToShortTimeString()) : $message"
+
+    if(-not (Test-Path $Path))
+    {
+        Set-Content -Path $Path -Value $Message -Force -PassThru
+    }
+    else
+    {
+        Add-Content -Path $Path -Value $Message -PassThru
+    }
+}
+
+$codeCoverageZip = 'https://ci.appveyor.com/api/projects/PowerShell/powershell-f975h/artifacts/CodeCoverage.zip'
+$testContentZip = 'https://ci.appveyor.com/api/projects/PowerShell/powershell-f975h/artifacts/tests.zip'
+$openCoverZip = 'https://ci.appveyor.com/api/projects/PowerShell/powershell-f975h/artifacts/OpenCover.zip'
+
+Write-LogPassThru -Message "codeCoverageZip: $codeCoverageZip"
+Write-LogPassThru -Message "testcontentZip: $testContentZip"
+Write-LogPassThru -Message "openCoverZip: $openCoverZip"
+
+$outputBaseFolder = "$env:Temp\CC"
+$null = New-Item -ItemType Directory -Path $outputBaseFolder -Force
+
+$openCoverPath = "$outputBaseFolder\OpenCover"
+$testPath = "$outputBaseFolder\tests"
+$psBinPath = "$outputBaseFolder\PSCodeCoverage"
+$openCoverTargetDirectory = "$outputBaseFolder\OpenCoverToolset"
+$outputLog = "$outputBaseFolder\CodeCoverageOutput.xml"
+$psCodeZip = "$outputBaseFolder\PSCode.zip"
+$psCodePath = "$outputBaseFolder\PSCode"
+
+try
+{
+    Write-LogPassThru -Message "Starting downloads."
+
+    Invoke-WebRequest -uri $codeCoverageZip -outfile "$outputBaseFolder\PSCodeCoverage.zip"
+    Invoke-WebRequest $testContentZip -outfile "$outputBaseFolder\tests.zip"
+    Invoke-WebRequest $openCoverZip -outfile "$outputBaseFolder\OpenCover.zip"
+
+    Write-LogPassThru -Message "Downloads complete."
+
+    Write-LogPassThru -Message "Starting expansion."
+
+    Expand-Archive -path "$outputBaseFolder\PSCodeCoverage.zip" -destinationpath "$psBinPath" -Force
+    Expand-Archive -path "$outputBaseFolder\tests.zip" -destinationpath $testPath -Force
+    Expand-Archive -path "$outputBaseFolder\OpenCover.zip" -destinationpath $openCoverPath -Force
+
+    ## Download Coveralls.net uploader
+    $coverallsToolsUrl = 'https://github.com/csMACnz/coveralls.net/releases/download/0.7.0/coveralls.net.0.7.0.nupkg'
+    $coverallsPath = "$outputBaseFolder\coveralls"
+
+    ## Saving the nupkg as zip so we can expand it.
+    Invoke-WebRequest $coverallsToolsUrl -outfile "$outputBaseFolder\coveralls.zip"
```

Comment:
fixed

---
In reply to: [94474891](https://github.com/PowerShell/PowerShell/pull/2918#discussion_r94474891) [](ancestors = 94474891)